pkg/addonfactory/addonfactory.go

```diff
@@ -87,13 +96,6 @@ func (f *AgentAddonFactory) WithTrimCRDDescription() *AgentAddonFactory {
 
 // BuildHelmAgentAddon builds a helm agentAddon instance.
 func (f *AgentAddonFactory) BuildHelmAgentAddon() (agent.AgentAddon, error) {
-	if f.scheme == nil {
```

Comment:
we cannot remove this because user may not add `WithScheme` for the `AgentAddonFactory`

Reply:
Just move it in `NewAgentAddonFactory` and `WithScheme` https://github.com/open-cluster-management-io/addon-framework/pull/103/files#diff-deb173045a5affa378b2b645e4f8b7be9e860e71641416b205be8f0de5c069ebR60